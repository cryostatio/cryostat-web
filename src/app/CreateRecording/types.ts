/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { EventTemplate, KeyValue } from '@app/Shared/Services/api.types';
import { ValidatedOptions } from '@patternfly/react-core';

export type EventTemplateIdentifier = Pick<EventTemplate, 'name' | 'type'>;

interface _FormBaseData {
  name: string;
  template?: EventTemplateIdentifier;
  labels: KeyValue[];
  continuous: boolean;
  archiveOnStop: boolean;
  restart: boolean;
  duration: number;
  durationUnit: number;
  skipDurationCheck?: boolean;
  maxAge: number;
  maxAgeUnit: number;
  maxSize: number;
  maxSizeUnit: number;
  toDisk: boolean;
}

interface _FormValidationData {
  nameValid: ValidatedOptions;
  labelsValid: ValidatedOptions;
  durationValid: ValidatedOptions;
}

export type CustomRecordingFormData = _FormBaseData & _FormValidationData;
