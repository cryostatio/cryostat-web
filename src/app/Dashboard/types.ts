/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FeatureLevel } from '@app/Shared/Services/service.types';
import { gridSpans, LabelProps } from '@patternfly/react-core';
import { Observable } from 'rxjs';

export interface Sized<T> {
  minimum: T;
  default: T;
  maximum: T;
}

export interface DashboardCardSizes {
  span: Sized<gridSpans>;
  height: Sized<number>;
}

export interface DashboardCardDescriptor {
  featureLevel: FeatureLevel;
  icon?: React.ReactNode;
  labels?: {
    content: string;
    color?: LabelProps['color'];
    icon?: React.ReactNode;
  }[];
  preview?: React.ReactNode;
  title: string;
  cardSizes: DashboardCardSizes;
  description: string;
  descriptionFull: JSX.Element | string;
  component: DashboardCardFC<DashboardCardTypeProps>;
  propControls: PropControl[];
  advancedConfig?: JSX.Element;
}
export type DashboardCardFC<P> = React.FC<P> & {
  cardComponentName: string;
};

export interface PropControlExtra {
  displayMapper?: (value: string) => string /* only has effect with 'select' PropControl kind */;
  min?: number;
  max?: number;
  [key: string]: unknown;
}

export interface PropControl {
  name: string;
  key: string;
  description: string;
  kind: 'boolean' | 'number' | 'string' | 'text' | 'select';
  values?: unknown[] | Observable<unknown>;
  defaultValue: unknown;
  extras?: PropControlExtra;
}

export interface DashboardCardTypeProps {
  span: number;
  dashboardId: number;
  isDraggable?: boolean;
  isResizable?: boolean;
  isFullHeight?: boolean;
  actions?: JSX.Element[];
}

export interface ValidationError {
  message: React.ReactNode;
}

export interface CardValidationResult {
  errors: ValidationError[];
  callForAction?: React.ReactNode;
}

export type LayoutTemplateFilter = 'Suggested' | 'Cryostat' | 'User-submitted';

export interface LayoutTemplateController {
  selectedTemplate: SelectedLayoutTemplate | undefined;
  setSelectedTemplate: (template: React.SetStateAction<SelectedLayoutTemplate | undefined>) => void;
  isUploadModalOpen: boolean;
  setIsUploadModalOpen: (isOpen: React.SetStateAction<boolean>) => void;
}

export interface CardConfig {
  id: string;
  name: string;
  span: gridSpans;
  props: object;
}

export type SerialCardConfig = Omit<CardConfig, 'id'>;

export const mockSerialCardConfig: SerialCardConfig = {
  name: 'Default',
  span: 12,
  props: {},
};
export interface DashboardLayout {
  name: string;
  cards: CardConfig[];
  favorite: boolean;
}

export type SerialDashboardLayout = Omit<DashboardLayout, 'name' | 'cards'> & { cards: SerialCardConfig[] };

// only name and vendor are needed to identify a template
export type LayoutTemplateRecord = Pick<LayoutTemplate, 'name' | 'vendor'>;

export enum LayoutTemplateVersion {
  'v2.3' = 'v2.3',
  'v2.4' = 'v2.4',
}

export enum LayoutTemplateVendor {
  BLANK = 'Blank',
  CRYOSTAT = 'Cryostat',
  USER = 'User-submitted',
}

export interface LayoutTemplate {
  name: string;
  description: string;
  cards: SerialCardConfig[];
  version: LayoutTemplateVersion;
  vendor: LayoutTemplateVendor;
}

export interface SelectedLayoutTemplate {
  template: LayoutTemplate;
  category: LayoutTemplateFilter;
}

export type SerialLayoutTemplate = Omit<LayoutTemplate, 'vendor'>;
