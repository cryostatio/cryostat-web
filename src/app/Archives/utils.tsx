/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { RecordingDirectory, Target } from '@app/Shared/Services/api.types';

export const includesDirectory = (arr: RecordingDirectory[], dir: RecordingDirectory): boolean => {
  return arr.some((t) => t.connectUrl === dir.connectUrl);
};

export const indexOfDirectory = (arr: RecordingDirectory[], dir: RecordingDirectory): number => {
  let index = -1;
  arr.forEach((d, idx) => {
    if (d.connectUrl === dir.connectUrl) {
      index = idx;
    }
  });
  return index;
};

export const getTargetFromDirectory = (dir: RecordingDirectory): Target => {
  return {
    connectUrl: dir.connectUrl,
    alias: dir.jvmId,
    labels: [],
    annotations: {
      cryostat: [],
      platform: [],
    },
  };
};
