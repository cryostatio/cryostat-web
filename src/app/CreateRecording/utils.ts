/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { TemplateType } from '@app/Shared/Services/api.types';
import { EventTemplateIdentifier } from './types';

export const RecordingNamePattern = /^[\w_]+$/;

export const DurationPattern = /^[1-9][0-9]*$/;

export const isRecordingNameValid = (name: string) => RecordingNamePattern.test(name);

export const isDurationValid = (duration: number) => DurationPattern.test(`${duration}`);

export const templateFromEventSpecifier = (eventSpecifier: string): EventTemplateIdentifier | undefined => {
  const regex = /^template=([a-zA-Z0-9]+)(?:,type=([a-zA-Z0-9]+))?$/im;
  if (eventSpecifier && regex.test(eventSpecifier)) {
    const parts = regex.exec(eventSpecifier);
    if (parts) {
      return {
        name: parts[1],
        type: parts[2] as TemplateType,
      };
    }
  }
  return undefined;
};
