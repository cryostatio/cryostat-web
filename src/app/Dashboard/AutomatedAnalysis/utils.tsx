/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AnalysisResult, Evaluation } from '@app/Shared/Services/api.types';
import { Stack, StackItem, Label, Title, Text } from '@patternfly/react-core';
import _ from 'lodash';

export const transformAADescription = (result: AnalysisResult): JSX.Element => {
  const format = (s: Evaluation): JSX.Element => {
    if (typeof s === 'string') {
      return <Text>{s}</Text>;
    }
    if (Array.isArray(s)) {
      return (
        <Stack>
          {s.map((e) => (
            <StackItem key={e.setting}>
              <Title headingLevel={'h6'}>{e.setting}</Title>
              <Label>
                {e.name}={e.value}
              </Label>
            </StackItem>
          ))}
        </Stack>
      );
    }
    throw `Unrecognized item: ${s}`;
  };
  return (
    <div>
      {Object.entries(result.evaluation || {}).map(([k, v]) =>
        v && v.length ? (
          <div key={k}>
            <span>
              <Title headingLevel={'h5'}>{_.capitalize(k)}</Title>
              {format(result.evaluation[k])}
            </span>
            <br />
          </div>
        ) : (
          <div key={k}></div>
        ),
      )}
    </div>
  );
};
