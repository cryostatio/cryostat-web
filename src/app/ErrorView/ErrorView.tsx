/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Button, EmptyState, EmptyStateBody, EmptyStateIcon, Title, StackItem, Stack } from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import * as React from 'react';

export const authFailMessage = 'Authentication failure';

export const missingSSLMessage = 'Bad Gateway';

export const isAuthFail = (message: string) => message === authFailMessage;
export interface ErrorViewProps {
  title: string | React.ReactNode;
  message: string | React.ReactNode;
  retryButtonMessage?: string;
  retry?: () => void;
}

export const ErrorView: React.FC<ErrorViewProps> = (props) => {
  return (
    <>
      <EmptyState>
        <EmptyStateIcon icon={ExclamationCircleIcon} color={'#a30000'} />
        <Title headingLevel="h4" size="lg">
          {props.title}
        </Title>
        <EmptyStateBody>
          <>
            <Stack>
              <StackItem>{props.message}</StackItem>
              {props.retry && (
                <StackItem>
                  <Button variant="link" onClick={props.retry}>
                    {props.retryButtonMessage || 'Retry'}
                  </Button>
                </StackItem>
              )}
            </Stack>
          </>
        </EmptyStateBody>
      </EmptyState>
    </>
  );
};
