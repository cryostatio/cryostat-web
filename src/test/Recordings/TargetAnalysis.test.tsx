/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import '@testing-library/jest-dom';
import { authFailMessage } from '@app/ErrorView/types';
import { DeleteActiveRecordings, DeleteOrDisableWarningType } from '@app/Modal/types';
import { ActiveRecordingsTable } from '@app/Recordings/ActiveRecordingsTable';
import {
  emptyActiveRecordingFilters,
  emptyArchivedRecordingFilters,
  TargetRecordingFilters,
} from '@app/Shared/Redux/Filters/RecordingFilterSlice';
import { RootState } from '@app/Shared/Redux/ReduxStore';
import {
  ActiveRecording,
  RecordingState,
  NotificationMessage,
  Target,
  keyValueToString,
  AggregateReport,
} from '@app/Shared/Services/api.types';
import { defaultServices, ServiceContext, Services } from '@app/Shared/Services/Services';
import { TargetService } from '@app/Shared/Services/Target.service';
import dayjs, { defaultDatetimeFormat } from '@i18n/datetime';
import { act, cleanup, screen, within } from '@testing-library/react';
import { NEVER, of, Subject } from 'rxjs';
import { basePreloadedState, DEFAULT_DIMENSIONS, render, renderSnapshot, resize, testT } from '../utils';
import { TargetAnalysis } from '@app/Recordings/TargetAnalysis';

const mockConnectUrl = 'service:jmx:rmi://someUrl';
const mockJvmId = 'id';
const mockTarget = {
  agent: false,
  connectUrl: mockConnectUrl,
  alias: 'fooTarget',
  jvmId: mockJvmId,
  labels: [],
  annotations: { cryostat: [], platform: [] },
};
const mockRecordingLabels = [
  {
    key: 'someLabel',
    value: 'someValue',
  },
];
const mockRecording: ActiveRecording = {
  id: 1,
  name: 'someRecording',
  downloadUrl: 'http://downloadUrl',
  reportUrl: 'http://reportUrl',
  metadata: { labels: mockRecordingLabels },
  startTime: 1234567890,
  state: RecordingState.RUNNING,
  duration: 1000, // 1000ms
  continuous: false,
  toDisk: false,
  maxSize: 0,
  maxAge: 0,
  remoteId: 998877,
};

const mockReport: AggregateReport = {
  lastUpdated: 1744213233,
  aggregate: {
    count: 1,
    max: 50.0,
  },
  data: [
    {
      key: 'rule',
      value: {
        evaluation: {
          explanation: 'because of test',
          solution: 'nothing',
          suggestions: [
            {
              name: 'try this',
              setting: 'A',
              value: '1234',
            },
          ],
          summary: 'we found something',
        },
        name: 'tests',
        topic: 'testsuite',
        score: 50.0,
      },
    },
  ],
};

const mockEmptyReport: AggregateReport = {
  lastUpdated: 1744213233,
  aggregate: {
    count: 0,
    max: 0.0,
  },
  data: [],
};

jest.mock('@app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCardList', () => {
  return {
    ...jest.requireActual('@app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCardList'),
    AutomatedAnalysisCardList: jest.fn(() => {
      return <div>AutomatedAnalysisCardList</div>;
    }),
  };
});

jest.mock('@app/Dashboard/AutomatedAnalysis/ClickableAutomatedAnalysisLabel', () => {
  return {
    ...jest.requireActual('@app/Dashboard/AutomatedAnalysis/ClickableAutomatedAnalysisLabel'),
    ClickableAutomatedAnalysisLabel: jest.fn(() => {
      return <div>ClickableAutomatedAnalysisLabel</div>;
    }),
    ClickableAutomatedAnalysisKey: jest.fn(() => {
      return <div>ClickableAutomatedAnalysisKey</div>;
    }),
  };
});

jest.mock('@app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters', () => {
  return {
    ...jest.requireActual('@app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters'),
    AutomatedAnalysisFilters: jest.fn(() => {
      return <div>AutomatedAnalysisFilters</div>;
    }),
  };
});

jest.spyOn(defaultServices.settings, 'datetimeFormat').mockReturnValue(of(defaultDatetimeFormat));
jest.spyOn(defaultServices.target, 'target').mockReturnValue(of(mockTarget));
jest.spyOn(defaultServices.api, 'getTargetActiveRecordings').mockReturnValue(of([mockRecording]));
jest.spyOn(defaultServices.api, 'getCurrentReportForTarget').mockReturnValue(of(mockReport));
jest.spyOn(defaultServices.api, 'sendRequest').mockReturnValue(of());
jest.spyOn(defaultServices.notificationChannel, 'messages').mockImplementation((category: string) => {
  switch (category) {
    case 'ReportSuccess':
      return of({
        meta: { category: 'ReportSuccess', type: { type: 'application', subtype: 'json' } },
        message: { jvmId: mockTarget.jvmId },
      });
    default:
      return of();
  }
});

describe('<TargetAnalysis />', () => {
  afterEach(cleanup);

  it('renders correctly', async () => {
    const tree = await renderSnapshot({
      routerConfigs: {
        routes: [
          {
            path: '/',
            element: <TargetAnalysis target={mockTarget} />,
          },
        ],
      },
    });
    expect(tree?.toJSON()).toMatchSnapshot();
  });

  it('renders loading state', async () => {
    jest.spyOn(defaultServices.api, 'getCurrentReportForTarget').mockRestore();
    jest.spyOn(defaultServices.api, 'getCurrentReportForTarget').mockReturnValue(NEVER);
    render({
      routerConfigs: {
        routes: [
          {
            path: '/',
            element: <TargetAnalysis target={mockTarget} />,
          },
        ],
      },
    });

    expect(screen.getByRole('progressbar', { name: 'Contents' })).toBeInTheDocument();
  });

  it('renders empty state', async () => {
    jest.spyOn(defaultServices.api, 'getCurrentReportForTarget').mockRestore();
    jest.spyOn(defaultServices.api, 'getCurrentReportForTarget').mockReturnValue(of(mockEmptyReport));
    render({
      routerConfigs: {
        routes: [
          {
            path: '/',
            element: <TargetAnalysis target={mockTarget} />,
          },
        ],
      },
    });

    expect(screen.getByText(testT('TargetAnalysis.REPORT_UNAVAILABLE'))).toBeInTheDocument();
    expect(screen.queryByText('ClickableAutomatedAnalysisLabel')).not.toBeInTheDocument();
  });
});
