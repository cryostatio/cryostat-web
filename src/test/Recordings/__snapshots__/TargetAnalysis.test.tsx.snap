// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<TargetAnalysis /> renders correctly 1`] = `
<div
  className="pf-v5-l-stack pf-m-gutter"
>
  <div
    className="pf-v5-l-stack__item"
  >
    <div
      style={
        {
          "display": "contents",
        }
      }
    >
      <p
        className=""
        data-ouia-component-id="OUIA-Generated-Text-2"
        data-ouia-component-type="PF5/Text"
        data-ouia-safe={true}
        data-pf-content={true}
      >
        Last update: Wednesday, April 9, 2025 3:40 PM
      </p>
    </div>
  </div>
  <div
    className="pf-v5-l-stack__item"
  >
    <div
      aria-label="Automated analysis toolbar"
      className="pf-v5-c-toolbar pf-m-full-height"
      data-ouia-component-id="OUIA-Generated-Toolbar-1"
      data-ouia-component-type="PF5/Toolbar"
      data-ouia-safe={true}
      id="automated-analysis-toolbar"
    >
      <div
        className="pf-v5-c-toolbar__content"
      >
        <div
          className="pf-v5-c-toolbar__content-section"
        >
          <div>
            AutomatedAnalysisFilters
          </div>
          <div
            className="pf-v5-c-toolbar__item"
          >
            <div
              className="pf-v5-c-check"
            >
              <input
                aria-invalid={false}
                checked={false}
                className="pf-v5-c-check__input"
                data-ouia-component-id="OUIA-Generated-Checkbox-1"
                data-ouia-component-type="PF5/Checkbox"
                data-ouia-safe={true}
                disabled={false}
                id="show-na-scores"
                name="show-na-scores"
                onChange={[Function]}
                required={false}
                style={
                  {
                    "alignSelf": "center",
                  }
                }
                type="checkbox"
              />
              <label
                className="pf-v5-c-check__label"
                htmlFor="show-na-scores"
              >
                Show N/A scores
              </label>
            </div>
          </div>
          <hr
            className="pf-v5-c-divider pf-m-vertical"
          />
          <div
            className="pf-v5-c-toolbar__item"
          >
            <div
              className="pf-v5-c-toggle-group"
              role="group"
            >
              <div
                className="pf-v5-c-toggle-group__item"
                onClick={[Function]}
              >
                <button
                  aria-label="grid-view"
                  aria-pressed={true}
                  className="pf-v5-c-toggle-group__button pf-m-selected"
                  id="grid-view-btn"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="pf-v5-c-toggle-group__text"
                  >
                    Grid view
                  </span>
                </button>
              </div>
              <div
                className="pf-v5-c-toggle-group__item"
                onClick={[Function]}
              >
                <button
                  aria-label="list-view"
                  aria-pressed={false}
                  className="pf-v5-c-toggle-group__button"
                  id="list-view-btn"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="pf-v5-c-toggle-group__text"
                  >
                    List view
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pf-v5-c-toolbar__content pf-m-hidden"
        hidden={true}
      >
        <div
          className="pf-v5-c-toolbar__group"
        />
      </div>
    </div>
  </div>
  <div
    className="pf-v5-l-stack__item"
  >
    <div
      style={
        {
          "display": "contents",
        }
      }
    >
      <small
        className=""
        data-ouia-component-id="OUIA-Generated-Text-3"
        data-ouia-component-type="PF5/Text"
        data-ouia-safe={true}
        data-pf-content={true}
      >
        Only showing analysis results with severity scores â‰¥ 0:
      </small>
    </div>
    <div
      className="pf-v5-c-slider automated-analysis-score-filter-slider automated-analysis-score-filter-slider-ok"
      style={
        {
          "--pf-v5-c-slider--value": "0%",
          "--pf-v5-c-slider__value--c-form-control--width-chars": 1,
        }
      }
    >
      <div
        className="pf-v5-c-slider__actions"
      >
        <div
          className="pf-v5-l-level pf-m-gutter"
        >
          <div>
            <small
              className=""
              data-ouia-component-id="OUIA-Generated-Text-4"
              data-ouia-component-type="PF5/Text"
              data-ouia-safe={true}
              data-pf-content={true}
            >
              Reset
              :
            </small>
          </div>
          <div>
            <button
              aria-disabled={false}
              aria-label="Reset score to 0"
              className="pf-v5-c-button pf-m-link pf-m-inline pf-m-small"
              data-ouia-component-id="OUIA-Generated-Button-link-1"
              data-ouia-component-type="PF5/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[Function]}
              role={null}
              type="button"
            >
              0
            </button>
          </div>
          <div>
            <button
              aria-disabled={false}
              aria-label="Reset score to 100"
              className="pf-v5-c-button pf-m-link pf-m-inline pf-m-small"
              data-ouia-component-id="OUIA-Generated-Button-link-2"
              data-ouia-component-type="PF5/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[Function]}
              role={null}
              type="button"
            >
              100
            </button>
          </div>
        </div>
      </div>
      <div
        className="pf-v5-c-slider__main"
      >
        <div
          className="pf-v5-c-slider__rail"
          onClick={[Function]}
        >
          <div
            className="pf-v5-c-slider__rail-track"
          />
        </div>
        <div
          aria-hidden="true"
          className="pf-v5-c-slider__steps"
        >
          <div
            className="pf-v5-c-slider__step pf-m-active"
            style={
              {
                "--pf-v5-c-slider__step--Left": "0%",
              }
            }
          >
            <div
              className="pf-v5-c-slider__step-tick"
            />
            <div
              className="pf-v5-c-slider__step-label"
            >
              0
            </div>
          </div>
          <div
            className="pf-v5-c-slider__step"
            style={
              {
                "--pf-v5-c-slider__step--Left": "12.5%",
              }
            }
          >
            <div
              className="pf-v5-c-slider__step-tick"
            />
            <div
              className="pf-v5-c-slider__step-label"
            >
              OK
            </div>
          </div>
          <div
            className="pf-v5-c-slider__step"
            style={
              {
                "--pf-v5-c-slider__step--Left": "25%",
              }
            }
          >
            <div
              className="pf-v5-c-slider__step-tick"
            />
            <div
              className="pf-v5-c-slider__step-label"
            >
              25
            </div>
          </div>
          <div
            className="pf-v5-c-slider__step"
            style={
              {
                "--pf-v5-c-slider__step--Left": "50%",
              }
            }
          >
            <div
              className="pf-v5-c-slider__step-tick"
            />
            <div
              className="pf-v5-c-slider__step-label"
            >
              WARNING
            </div>
          </div>
          <div
            className="pf-v5-c-slider__step"
            style={
              {
                "--pf-v5-c-slider__step--Left": "75%",
              }
            }
          >
            <div
              className="pf-v5-c-slider__step-tick"
            />
            <div
              className="pf-v5-c-slider__step-label"
            >
              75
            </div>
          </div>
          <div
            className="pf-v5-c-slider__step"
            style={
              {
                "--pf-v5-c-slider__step--Left": "87.5%",
              }
            }
          >
            <div
              className="pf-v5-c-slider__step-tick"
            />
            <div
              className="pf-v5-c-slider__step-label"
            >
              CRITICAL
            </div>
          </div>
          <div
            className="pf-v5-c-slider__step"
            style={
              {
                "--pf-v5-c-slider__step--Left": "100%",
              }
            }
          >
            <div
              className="pf-v5-c-slider__step-tick"
            />
            <div
              className="pf-v5-c-slider__step-label"
            >
              100
            </div>
          </div>
        </div>
        <div
          aria-disabled={false}
          aria-label="Value"
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          aria-valuetext="0"
          className="pf-v5-c-slider__thumb"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          role="slider"
          tabIndex={0}
        />
      </div>
      <div
        className="pf-v5-c-slider__value"
      >
        <div
          className="pf-v5-c-input-group"
        >
          <div
            className="pf-v5-c-input-group__item pf-m-fill"
          >
            <span
              className="pf-v5-c-form-control"
            >
              <input
                aria-invalid={false}
                aria-label="Slider value input"
                data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                data-ouia-component-type="PF5/TextInput"
                data-ouia-safe={true}
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                required={false}
                type="number"
                value={0}
              />
            </span>
          </div>
          <div
            className="pf-v5-c-input-group__item pf-m-box"
          >
            <span
              className="pf-v5-c-input-group__text"
            >
              Score
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pf-v5-l-stack__item pf-m-fill"
  >
    <div
      className="pf-v5-l-grid"
    >
      <div
        className="pf-v5-l-grid__item pf-m-3-col automated-analysis-grid-item"
      >
        <div
          className="pf-v5-c-label-group automated-analysis-topic-label-groups pf-m-category pf-m-vertical"
        >
          <div
            className="pf-v5-c-label-group__main"
          >
            <span
              aria-hidden="true"
              className="pf-v5-c-label-group__label"
              id="pf-random-id-0"
            >
              testsuite
            </span>
            <ul
              aria-labelledby="pf-random-id-0"
              className="pf-v5-c-label-group__list"
              role="list"
            >
              <li
                className="pf-v5-c-label-group__list-item"
              >
                <div>
                  ClickableAutomatedAnalysisLabel
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
